core/amber/src/main/scala/edu/uci/ics/texera/web/service/ExecutionResultService.scala

```diff
@@ -63,12 +62,23 @@ object ExecutionResultService {
     * Produces the WebResultUpdate to send to frontend from a result update from the engine.
     */
   private def convertWebResultUpdate(
-      sink: ProgressiveSinkOpDesc,
+      workflowIdentity: WorkflowIdentity,
+      physicalOps: List[PhysicalOp],
       oldTupleCount: Int,
       newTupleCount: Int
   ): WebResultUpdate = {
+    val outputMode = physicalOps
+      .flatMap(op => op.outputPorts)
+      .filter({
+        case (portId, (port, links, schema)) => !portId.internal
+      })
+      .map({
+        case (portId, (port, links, schema)) => port.mode
+      })
+      .head
```

Comment:
better to add a todo here saying we will extend this to multi-output ports.

Reply:
done